src/main/kotlin/com/exactpro/th2/codec/api/impl/ReportingContext.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.codec.api.impl
+
+import com.exactpro.th2.codec.api.IReportingContext
+
+class ReportingContext : IReportingContext {
+    private val _warnings = arrayListOf<String>()
+    val warnings: List<String>
+        get() = _warnings
```

Comment:
```suggestion
    val warnings: List<String> by ::_warnings
```

Reply:
It looks prettier but it will add an additional field with property delegate to the class. I think the pretty look does not worth it